client/src/sim.rs

```diff
@@ -75,6 +80,13 @@ impl Sim {
         self.movement_input = movement_input;
     }
 
+    /// Prepares no_clip to be toggled after the next step. We avoid toggling it immediately, as
+    /// that would cause a discontinuity when predicting the player's position within a given step,
+    /// causing an undesirable jolt.
```

Comment:
nit: doc-comments should only contain information relevant to the caller, whereas information on the (reasoning behind the) internals should be relegated to regular `//` comments, and perhaps placed lexically inside the function.

Reply:
Since the only doc comments that are relevant to the caller are that the function toggles no_clip, I opted to remove them entirely.